### ISSUE TITLE
Flaky test: test_get_issue_number_with_sort_order

### ISSUE THREAD

**Issue by lucyb** [opensafely-core/job-server#4501]
I've had a couple of intermittent test failures today ([example](https://github.com/opensafely-core/job-server/actions/runs/10163678642/job/28107189006?pr=4500)), which have resulted in the same test failure:

```
  File "/app/tests/verification/test_github.py", line 90, in test_get_issue_number_with_sort_order
    assert default_order == latest
AssertionError: assert 8901 == 8902
=========================== short test summary info ============================
FAILED tests/verification/test_github.py::test_get_issue_number_with_sort_order
============= 1 failed, 2116 passed, 14 subtests passed in 30.53s ==============
error: Recipe `test` failed with exit code 1
error: Recipe `docker-test` failed on line 268 with exit code 1
Error: Process completed with exit code 1.
```

I assume that any new issues take a while to propagate through GitHub's systems, so the API can return inconsistent results for a short period of time. However, I've not investigated and could be wrong about that.

**Comment by StevenMaude** [comment]
Is it also possible there's a chance of a race condition if multiple test runs are simultaneous? 

From looking (very quickly), I'm right in thinking that this really creates issues in a real repository? (I can't see it, because of the permissions being in a different organization: `opensafely-testing`.)

[Your run](https://github.com/opensafely-core/job-server/actions/runs/10161042678) that failed was about the same time as [my run](https://github.com/opensafely-core/job-server/actions/runs/10161042678) that passed.

It might be another hint that we should move the verification tests to run on a schedule, not as part of the regular CI (see discussion in #4434).

**Comment by lucyb** [comment]
> [Your run](https://github.com/opensafely-core/job-server/actions/runs/10161053776/attempts/1) that failed was about the same time as [my run](https://github.com/opensafely-core/job-server/actions/runs/10161042678) that passed.

Oh good spot. Yes, I think you're right and I agree about implementing https://github.com/opensafely-core/job-server/issues/4434.

**Comment by lucyb** [comment]
I'm going to close this issue with the expectation that we'll fix it in #4434.
